File: backend/src/main/java/com/example/demo/bean/ProblemDiagram.java
package com.example.demo.bean;

import java.util.List;

public class ProblemDiagram {
	private String title;	//文件名
	private ContextDiagram contextDiagram;	//上下文图
	private List<Requirement> requirementList;	//需求列表
	private List<Constraint> constraintList;	//约束列表
	private List<Reference> referenceList;	//引用列表
	
	public String getTitle() {
		return title;
	}
	public void setTitle(String title) {
		this.title = title;
	}
	public ContextDiagram getContextDiagram() {
		return contextDiagram;
	}
	public void setContextDiagram(ContextDiagram contextDiagram) {
		this.contextDiagram = contextDiagram;
	}
	public List<Requirement> getRequirementList() {
		return requirementList;
	}
	public void setRequirementList(List<Requirement> requirementList) {
		this.requirementList = requirementList;
	}
	public List<Constraint> getConstraintList() {
		return constraintList;
	}
	public void setConstraintList(List<Constraint> constraintList) {
		this.constraintList = constraintList;
	}
	public List<Reference> getReferenceList() {
		return referenceList;
	}
	public void setReferenceList(List<Reference> referenceList) {
		this.referenceList = referenceList;
	}
}

File: frontend/src/app/entity/ProblemDiagram.ts
import { ContextDiagram } from './ContextDiagram'
import { Requirement } from './Requirement'
import { Constraint } from './Constraint'
import { Reference } from './Reference'

export class ProblemDiagram {
	title: string
	contextDiagram: ContextDiagram
	requirementList: Requirement[]
	constraintList: Constraint[]
	referenceList: Reference[]
	static copyProblemDiagram(old:ProblemDiagram,contextDiagram: ContextDiagram):ProblemDiagram{
		let new1 = new ProblemDiagram()
		new1.title = old.title
		new1.contextDiagram = contextDiagram
		new1.requirementList = new Array<Requirement>()
		if(old.requirementList!=null){
			for(let oldItem of old.requirementList){
				let item = Requirement.copyRequirement(oldItem)
				new1.requirementList.push(item)
			}			
		}
		new1.constraintList = new Array<Constraint>()
		if(old.constraintList!=null){
			for(let oldItem of old.constraintList){
				let item = Constraint.copyConstraint(oldItem)
				new1.constraintList.push(item)
			}			
		}
		new1.referenceList = new Array<Reference>()
		if(old.referenceList!=null){
			for(let oldItem of old.referenceList){
				let item = Reference.copyReference(oldItem)
				new1.referenceList.push(item)
			}			
		}			
		return new1
	}
}